### PR TITLE
Logging explanations

### PR DESCRIPTION
**Pull request by haaakon**
Adding some helper methods to make it easier posting logs with a understandable format

### CODE REVIEW

File: Digipost/Logger.swift [hunk 1]
```diff
@@ -25,39 +25,50 @@ class Logger {
...
+        let parameters = [ Logger.Constants.severity : severity.rawValue, Logger.Constants.message : message ]
+        APIClient.sharedClient.postLog(uri: Logger.Constants.uri, parameters: parameters)
     }
 
-
-
 }
 
- func DLog(message: String, function: String = __FUNCTION__) {
+func DLog(message: String, function: String = __FUNCTION__) {
```
Comment (lillesand) [inline Digipost/Logger.swift:71]: Maybe we should rename from DLog to `debugLog` or something like that? Maybe `localDebugLog`…

File: Digipost/APIClient.swift [hunk 2]
```diff
@@ -427,9 +418,9 @@ class APIClient : NSObject, NSURLSessionTaskDelegate, NSURLSessionDelegate, NSUR
         let baseUri = __SERVER_URI__
         let completeUri = "\(baseUri)\(uri)"
         let task = self.urlSessionTaskWithNoAuthorizationHeader(httpMethod.post, url: completeUri, parameters: parameters, success: { () -> Void in
-            DLog("successfully sent log")
+            DLog("Successfully sent: \(parameters)")
```
Comment (lillesand) [inline Digipost/APIClient.swift]: Suggest we use something slightly more descriptive like "Successfully sent log with parameters …"

Similar for the error.

File: Digipost/APIClient.swift [hunk 3]
```diff
@@ -471,7 +462,8 @@ class APIClient : NSObject, NSURLSessionTaskDelegate, NSURLSessionDelegate, NSUR
             if (lowerLevelOAuthToken != nil) {
                 validate(oAuthToken: lowerLevelOAuthToken, validationSuccess: validationSuccess, failure: failure)
             } else {
-                Logger.dpostLogError("User revoked token and had no lower level token to fall back on")
+                Logger.dpostLogError("User revoked OAuth token and had no lower level token to fall back on", location: "Unknown, anywhere where there is a request to digipost API", UI: "User gets logged out", cause: "Something wrong with storing OAuth Tokens")
```
Comment (lillesand) [inline Digipost/APIClient.swift:465]: :+1:

**Comment by lillesand** [comment]
I think this is a really good change. This also gives us a better framework for tracking errors in the app for the future. Good job!

Review the comments and merge when done.

File: Digipost/Logger.swift (hunk 1 above)
Comment (haaakon) [inline Digipost/Logger.swift:71]: DLog is actually a common unofficial standard way of logging if DEBUG is set in the iOS world
http://stackoverflow.com/questions/9659763/difference-between-nslog-and-dlog
Alot of projects are setup with DLog

File: Digipost/APIClient.swift (hunk 2 above)
Comment (haaakon) [inline Digipost/APIClient.swift]: commit: 9ccfefc3ace96a44be81598fc8f76c164c0c67b6

**Pull request merged**